test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1304 @@
+		Name:               tcInfo.Name,
+		AllocatedResources: alloc,
+	}
+	return tc, tcStatus
+}
+
+func makeTestPod(ns, name, timeStamp string, tcInfo []TestContainerInfo) *v1.Pod {
+	var testContainers []v1.Container
+	var podOS *v1.PodOS
+
+	for _, ci := range tcInfo {
+		tc, _ := makeTestContainer(ci)
+		testContainers = append(testContainers, tc)
+	}
+
+	podOS = &v1.PodOS{Name: v1.Linux}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+			Labels: map[string]string{
+				"time": timeStamp,
+			},
+		},
+		Spec: v1.PodSpec{
+			OS:            podOS,
+			Containers:    testContainers,
+			RestartPolicy: v1.RestartPolicyOnFailure,
+		},
+	}
+	return pod
+}
+
+func verifyPodResizePolicy(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	cMap := make(map[string]*v1.Container)
+	for i, c := range pod.Spec.Containers {
+		cMap[c.Name] = &pod.Spec.Containers[i]
+	}
+	for _, ci := range tcInfo {
+		gomega.Expect(cMap).Should(gomega.HaveKey(ci.Name))
+		c := cMap[ci.Name]
+		tc, _ := makeTestContainer(ci)
+		gomega.Expect(tc.ResizePolicy).To(gomega.Equal(c.ResizePolicy))
+	}
+}
+
+func verifyPodResources(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	cMap := make(map[string]*v1.Container)
+	for i, c := range pod.Spec.Containers {
+		cMap[c.Name] = &pod.Spec.Containers[i]
+	}
+	for _, ci := range tcInfo {
+		gomega.Expect(cMap).Should(gomega.HaveKey(ci.Name))
+		c := cMap[ci.Name]
+		tc, _ := makeTestContainer(ci)
+		gomega.Expect(tc.Resources).To(gomega.Equal(c.Resources))
+	}
+}
+
+func verifyPodAllocations(pod *v1.Pod, tcInfo []TestContainerInfo) error {
+	cStatusMap := make(map[string]*v1.ContainerStatus)
+	for i, c := range pod.Status.ContainerStatuses {
+		cStatusMap[c.Name] = &pod.Status.ContainerStatuses[i]
+	}
+
+	for _, ci := range tcInfo {
+		gomega.Expect(cStatusMap).Should(gomega.HaveKey(ci.Name))
+		cStatus := cStatusMap[ci.Name]
+		if ci.Allocations == nil {
+			if ci.Resources != nil {
+				alloc := &ContainerAllocations{CPUAlloc: ci.Resources.CPUReq, MemAlloc: ci.Resources.MemReq}
+				ci.Allocations = alloc
+				defer func() {
+					ci.Allocations = nil
+				}()
+			}
+		}
+
+		_, tcStatus := makeTestContainer(ci)
+		if !cmp.Equal(cStatus.AllocatedResources, tcStatus.AllocatedResources) {
+			return fmt.Errorf("failed to verify Pod allocations, allocated resources not equal to expected")
+		}
+	}
+	return nil
+}
+
+func verifyPodStatusResources(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	csMap := make(map[string]*v1.ContainerStatus)
+	for i, c := range pod.Status.ContainerStatuses {
+		csMap[c.Name] = &pod.Status.ContainerStatuses[i]
+	}
+	for _, ci := range tcInfo {
+		gomega.Expect(csMap).Should(gomega.HaveKey(ci.Name))
+		cs := csMap[ci.Name]
+		tc, _ := makeTestContainer(ci)
+		gomega.Expect(tc.Resources).To(gomega.Equal(*cs.Resources))
+	}
+}
+
+func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework, pod *v1.Pod, tcInfo []TestContainerInfo) error {
```

Comment:
ditto

Reply:
Will try and comeback with a commit GinkgoHelper, thanks for sharing

Reply:
Done